metrics/lib/json.bash

```diff
@@ -129,6 +131,43 @@ metrics_json_end_of_system() {
 	system_index=$(( ${#json_result_array[@]}))
 }
 
+# Add a array of number of physical sockets and hypervisor per node within the Kubernetes cluster
+metrics_json_num_sockets_node(){
+
+	metrics_json_start_nested_array
+
+	# grab pods in the stats daemonset
+	while read -u 3 name node; do
+		# gather node information about number of sockets and its hypervisor if any
+		local num_sockets=$(kubectl exec -ti $name -- sh -c "cat /proc/cpuinfo | grep 'physical id' | sort -u | wc -l" | sed 's/\r//')
+		local node_hypervisor=false
+		kubectl exec -ti $name -- sh -c "grep -q ^flags.*\ hypervisor\  /proc/cpuinfo"
```

Comment:
This was not tested on the non-hypervisor case. On baremetal, k8s_scaling.sh fails at this line. We have our scripts setup to fail on any non-zero return value.  

You need something like to allow the error:
```
set +e
kubectl exec -ti $name -- sh -c "grep -q ^flags.*\ hypervisor\  /proc/cpuinfo"

if [ $? -eq 0 ]; then
     node_hypervisor=true
fi
set -e

Reply:
Done.

I messed up the previous test, sorry about that.